lib/HLSL/HLSignatureLower.cpp

```diff
@@ -1145,10 +1145,27 @@ void HLSignatureLower::GenerateDxilInputsOutputs(DXIL::SignatureKind SK) {
       bI1Cast = true;
       Ty = i32Ty;
     }
-    if (!hlslOP->IsOverloadLegal(opcode, Ty)) {
+  
+    if (!Ty || !hlslOP->IsOverloadLegal(opcode, Ty)) {
       std::string O;
       raw_string_ostream OSS(O);
-      Ty->print(OSS);
+      if (Ty)
+        Ty->print(OSS);
+      else {
+        auto Ty_iterator = HLM.GetEntryFunction()->getArgumentList().begin();
+        int argCount = 0;
+        llvm::Type *generic_Ty = nullptr;
+        while (Ty_iterator != HLM.GetEntryFunction()->getArgumentList().end()) {
+          if (argCount == i) {
+            generic_Ty = (*Ty_iterator).getType();
+            break;
+          }
+          Ty_iterator++;
+          argCount += 1;
+        }
+        generic_Ty->print(OSS);
```

Comment:
I suspect generic_Ty might still be null, which is causing the crash in automated testing. As we discussed, I think it's acceptable to do without the type for this error since it gives sufficient indication of where the problem is.